master/lustregittagpoller.py

```diff
@@ -0,0 +1,224 @@
+# -*- python -*-
+# ex: set syntax=python:
+
+import itertools
+import os
+import re
+import urllib
+
+from twisted.internet import defer
+from twisted.internet import utils
+from twisted.python import log
+
+from buildbot import config
+from buildbot.changes import base 
+from buildbot.util import epoch2datetime
+from buildbot.util.state import StateMixin
+
+class LustreTagPoller(base.PollingChangeSource, StateMixin):
+
+    """ This source will poll a remote git repo for
```

Comment:
Please read the python [Docstring Conventions](https://www.python.org/dev/peps/pep-0257/) and abide by them.  More explanation is definitely called for in this instance.